xarray/core/_reductions.py

```diff
@@ -26,6 +26,1811 @@ def reduce(
         ...
 
 
+class DataArrayReduce(Protocol):
```

Comment:
Could move these definitions to `types.py`?

Reply:
As you wish — it might be easier to keep the auto-gen content together though?